src/library/scala/collection/immutable/VectorMap.scala

```diff
@@ -115,22 +116,26 @@ final class VectorMap[K, +V] private[immutable] (
         (this eq map) ||
           (this.size == map.size) && {
             try {
-              var i = 0
               val _size = size
-              while (i < _size) {
-                val k = fields(i)
-
-                map.get(k) match {
-                  case Some(value) =>
-                    if (!(value == underlying(k)._2)) {
-                      return false
-                    }
-                  case None =>
-                    return false
+              @tailrec
+              def loop(i: Int): Boolean = {
```

Comment:
Ugh, this is almost twice as long and I find it harder to follow as well.  Is this really a good case to use a tail-recursive method?  (Or is there a simpler refactor?)

Reply:
Hmmm, use if-expressions? 🤔 

<https://github.com/xuwei-k/scala-1/commit/94d937c3af264132dd87671224bd88b309fe43a8?w=1>